### ISSUE TITLE
Log current learning rate during training for OneCycleLR visibility

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#19]
### Summary
The OneCycleLR scheduler is the primary training dynamic in this project — it ramps the learning rate up then decays it over the full training run. However, the current LR is never logged, making it impossible to inspect the schedule in TensorBoard/CSV logs or verify that the scheduler is working correctly.

### Current state
`configure_optimizers()` in `src/models/multi_head_resnet.py` sets up OneCycleLR correctly, but `training_step()` only logs `train_loss` and per-feature losses.

### Proposed change
Add a `LearningRateMonitor` callback when constructing the Trainer (in the notebook), or log LR directly in `training_step`:

**Option A — Trainer callback (recommended, no model changes needed):**
```python
from pytorch_lightning.callbacks import LearningRateMonitor

trainer = pl.Trainer(
    ...,
    callbacks=[LearningRateMonitor(logging_interval='step')],
)
```

**Option B — Log inside training_step:**
```python
opt = self.optimizers()
self.log('lr', opt.param_groups[0]['lr'], on_step=True, prog_bar=False)
```

### Why it matters
With a 20–30 epoch Colab run and OneCycleLR, the LR curve shape (warm-up → peak → decay) is the main signal that the scheduler is working. Without logging it, debugging convergence issues is significantly harder.

### Files affected
- `notebooks/set_card_classifier_main.ipynb` (Option A — preferred)
- Or `src/models/multi_head_resnet.py` (Option B)
